src/stream/mod.rs

```diff
@@ -148,33 +299,118 @@ impl Stream {
     /// This will download the video to <video_id>.mp4 in the current working directory.
     #[inline]
     pub async fn download(&self) -> Result<PathBuf> {
+        self.download_callback(None).await
+    }
+
+    /// Attempts to downloads the [`Stream`]s resource.
+    /// This will download the video to <video_id>.mp4 in the current working directory.
+    /// Takes an Callback
+    #[inline]
+    pub async fn download_callback(&self, callback: Option<Callback>) -> Result<PathBuf> {
```

Comment:
I'm not sure if those `*_callback` functions should have an `#[doc(cfg(feature = "callback"))]` flag even though they are callable without this feature.